include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -350,6 +350,14 @@ void aws_s3_meta_request_init_signing_date_time_default(
     struct aws_s3_meta_request *meta_request,
     struct aws_date_time *date_time);
 
+AWS_S3_API
+void aws_s3_meta_request_sign_request_default_impl(
+    struct aws_s3_meta_request *meta_request,
+    struct aws_s3_request *request,
+    aws_signing_complete_fn *on_signing_complete,
+    void *user_data,
+    bool force_regular_signing);
```

Comment:
`regular_signing` is confusing here, since we have three different types of signing algorithms and possibly more in the future. Might be simpler to just move this check to `s_meta_request_resolve_signing_config` and set the appropriate algorithm?

Reply:
I thought of adding the checking to `s3_meta_request.c`, but I think we want to keep the request type information limits to the separate type of meta request files. So, I'd like to keep the checking for different operations within `s3_copy_object.c`. I think the bool is enough for our current use case, if we ever want to support more complicated way to resolve the signing algorithms, we can think of something else later (Maybe another v function to resolve the signing configs?) And they are all private, which gives us flexibility to change the API in the future.

Reply:
We can rename `force_regular_signing` to something better. The Endpoints ruleset has the `disable_s3_express_session_auth` parameter; we can have something similar here like disable_s3_express_signing.